hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -178,6 +185,22 @@ public NodeEngineImpl(Node node) {
         }
     }
 
+    private void checkMapMergePolicies(Node node) {
```

Comment:
Looks good. But I'm wondering, and I know this config validation is somewhat scattered across the codebase, is this the right place to check the map policies? Is there some other place where we validate config? Did we do it at parse time?

It's a long-standing issue, but I always wanted to have a central `validateConfig` method, and have us call it from the parsers, startup procedure, wherever.

Reply:
We have validation checks in Node object constructor, i followed a similar way and put this one in NodeEngineImpl. Maybe we can have a better place but this seems an easy and ok place to put it, also we need config-class-loader to do checks and we don't have it at parse time.

Reply:
OK, thanks. And how about moving it to `Node` constructor, since you say there is already some config validation there?

Reply:
`SplitBrainMergePolicyProvider` is created in `NodeEngineImpl`, this was the other reason, didn't want to move code much.

Reply:
Ok, thanks.